typing/ctype.mli

```diff
@@ -283,9 +284,26 @@ val unify_gadt:
 val unify_var: Env.t -> type_expr -> type_expr -> unit
         (* Same as [unify], but allow free univars when first type
            is a variable. *)
-val filter_arrow: Env.t -> type_expr -> arg_label -> type_expr * type_expr
-        (* A special case of unification with [l:'a -> 'b].  Raises
-           [Filter_arrow_failed] instead of [Unify]. *)
+
+type filtered_arrow =
+  { ty_arg : type_expr;
+    ty_ret : type_expr;
+  }
+
+val filter_arrow: Env.t -> type_expr -> arg_label -> force_tpoly:bool ->
+        filtered_arrow
+        (* A special case of unification with [l:'a -> 'b]. If
+           [force_poly] is false then the usual invariant that the
+           argument type be a [Tpoly] node is not enforced. Raises
+           [Filter_arrow_failed] instead of [Unify].  *)
+val filter_mono: type_expr -> type_expr
+        (* A special case of unification (with Tpoly('a, [])). Can
+           only be called on [Tpoly] nodes. Raises [Filter_mono_failed]
+           instead of [Unify] *)
```

Comment:
The doc string here doesn't match the behaviour of `filter_mono`, namely in the case of `filter_mono 'b`, we do not unify `'b` with `Tpoly('a, [])`